src/bridge/MatrixHandler.ts

```diff
@@ -1046,13 +1052,55 @@ export class MatrixHandler {
 
         let cacheBody = ircAction.text;
 
-        // special handling for replies
-        if (event.content["m.relates_to"] && event.content["m.relates_to"]["m.in_reply_to"]) {
-            const eventId = event.content["m.relates_to"]["m.in_reply_to"].event_id;
-            const reply = await this.textForReplyEvent(event, eventId, ircRoom);
-            if (reply !== null) {
-                ircAction.text = reply.formatted;
-                cacheBody = reply.reply;
+        // special handling for replies and threads
+        if (event.content["m.relates_to"]) {
+            if (event.content["m.relates_to"]["m.in_reply_to"]) {
+                const eventId = event.content["m.relates_to"]["m.in_reply_to"].event_id;
+                const reply = await this.textForReplyEvent(event, eventId, ircRoom);
+                if (reply !== null) {
+                    ircAction.text = reply.formatted;
+                    cacheBody = reply.reply;
+                }
+            } else if (event.content["m.relates_to"]?.rel_type === "io.element.thread") {
+                const threadId = event.content["m.relates_to"].event_id;
+                let lastReplyId = this.lastMessageInThreadCache.get(threadId);
+                if (!lastReplyId) {
+                    try {
+                        const getThread = async (from?: string) => this.ircBridge.getAppServiceBridge().getIntent().matrixClient.doRequest(
```

Comment:
Might be worth chucking a PR at the bot sdk too, even if it will come too late for this PR.